clients/ts/signalr/src/HubConnectionBuilder.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { HttpConnection, IHttpConnectionOptions } from "./HttpConnection";
+import { HubConnection, JsonHubProtocol } from "./HubConnection";
+import { IHubProtocol } from "./IHubProtocol";
+import { ILogger, LogLevel } from "./ILogger";
+import { NullLogger } from "./Loggers";
+import { ConsoleLogger } from "./Utils";
+
+export class HubConnectionBuilder {
+    private protocol: any;
+    private httpConnectionOptions: IHttpConnectionOptions;
+    private url: string;
+    private logger: ILogger;
+
+    public configureLogging(logLevel: LogLevel): HubConnectionBuilder;
+    public configureLogging(logger: ILogger): HubConnectionBuilder;
+    public configureLogging(logLevelOrLogger: LogLevel | ILogger): HubConnectionBuilder {
```

Comment:
Are these parameter names idiomatic TS?

Reply:
I'll do a little looking to check but one thing to note is that when you do overloads like this in TypeScript, the actual parameter names (the signature that actually has a body) don't actually appear to the user in IntelliSense:

![image](https://user-images.githubusercontent.com/7574/38899516-f0a3138a-424c-11e8-8210-cd129c176324.png)
![image](https://user-images.githubusercontent.com/7574/38899535-fbf793f0-424c-11e8-9901-270833b6dff0.png)